### PR TITLE
add rna samples, gtf and refflat

### PR DESCRIPTION
**Pull request by DavyCats**
### Description

_Give your PR a **concise** yet **descriptive** title_

----

### Checklist (never delete this)

- [ ] Each method/class/object/trait should have scaladocs
- [ ] Added methods are unit tested
- [ ] Test coverage may not drop
- [ ] Documentation should be updated if required

### CODE REVIEW

File: src/main/scala/nl/biopet/utils/biowdl/references/Reference.scala [hunk 1]
```diff
@@ -59,4 +59,6 @@ trait Reference {
   def starGenomeDir: Option[File] = None
   def hisat2Index: Option[String] = None
 
+  def referenceGtf: Option[File] = None
```
Comment (ffinfo) [inline src/main/scala/nl/biopet/utils/biowdl/references/Reference.scala]: This should not be part of reference, this 1 reference can have multiple annotations.
Better make a trait Annotation like this:
```scala
trait Annotation extends Reference {
  //Do things
}
```

File: src/main/scala/nl/biopet/utils/biowdl/samples/Rna3.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                 "lib1",
+                 "rg1",
+                 Map("R1" -> fixtureFile("samples", "rna3", "R1.fq.gz"),
+                     "R1_md5" -> "2e53e727dad4cbbfedadeb29404ca0d3"))
+}
+
+trait Rna3PairedEnd extends Rna3SingleEnd {
+  override def samples: Map[String, Sample] =
+    addReadgroup(super.samples,
+                 "wgs1",
```
Comment (ffinfo) [inline src/main/scala/nl/biopet/utils/biowdl/samples/Rna3.scala]: Wrong sample?

**Review by ffinfo — CHANGES_REQUESTED** [review]

File: src/main/scala/nl/biopet/utils/biowdl/samples/Rna3.scala (hunk 2 above)
Comment (DavyCats) [reply src/main/scala/nl/biopet/utils/biowdl/samples/Rna3.scala]: oops :sweat_smile:

File: src/main/scala/nl/biopet/utils/biowdl/annotations/Annotation.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package nl.biopet.utils.biowdl.annotations
+
+import java.io.File
+
+import nl.biopet.utils.biowdl.references.Reference
+
+trait Annotation extends Reference {
+  def referenceGtf: Option[File] = None
```
Comment (ffinfo) [inline src/main/scala/nl/biopet/utils/biowdl/annotations/Annotation.scala]: No implementation needed here (so remove ' = None' ;) )

File: src/main/scala/nl/biopet/utils/biowdl/annotations/TestAnnotation.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package nl.biopet.utils.biowdl.annotations
+
+import java.io.File
+
+import nl.biopet.utils.biowdl.fixtureFile
+
+trait TestAnnotation extends Annotation {
+  override def referenceGtf: Option[File] =
```
Comment (ffinfo) [inline src/main/scala/nl/biopet/utils/biowdl/annotations/TestAnnotation.scala]: After comment above, no override needed anymore

**Review by ffinfo — CHANGES_REQUESTED** [review]

File: src/main/scala/nl/biopet/utils/biowdl/annotations/TestAnnotation.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.utils.biowdl.annotations
+
+import java.io.File
+
+import nl.biopet.utils.biowdl.fixtureFile
+
+trait TestAnnotation extends Annotation {
```
Comment (ffinfo) [inline src/main/scala/nl/biopet/utils/biowdl/annotations/TestAnnotation.scala]: should be bound to a reference, can you add `with TestReference`?

**Review by ffinfo — CHANGES_REQUESTED** [review]

**Review by ffinfo — APPROVED** [review]

**Pull request merged**